packages/types/src/Text.ts

```diff
@@ -43,7 +43,7 @@ export default class Text extends String implements Codec {
    * @description The length of the value when encoded as a Uint8Array
    */
   get encodedLength (): number {
-    return this.length + Compact.encodeU8a(this.length).length;
+    return this.toU8a().length;
```

Comment:
I am slightly worry about the performance impact of this, it should be possible to do it more efficiently

Reply:
It only gets used as part of the checking when decoding to determine offsets. So yes, impact while decoding metadata.